### DISCUSSION TITLE
Nextra 2.0.0 (beta.2)

### DISCUSSION THREAD

**Discussion by shuding** [shuding/nextra#257] [General]
Since 2.0.0, version numbers of the Nextra core and official themes (blog, docs) will be aligned. This release includes:
- `nextra-theme-blog@2.0.0-beta.2`
- `nextra-theme-docs@2.0.0-beta.2`
- `nextra@2.0.0-beta.2`

You can upgrade with `yarn add nextra@beta` or `npm install nextra@beta`.

### MDX v2

Nextra 2.0.0 will use MDX 2.0.0, please check the corresponding [migrating guide](https://mdxjs.com/migrating/v2) and [blog post](https://mdxjs.com/blog/v2) if you are interested.

### Highlighting Lines

**Before v2**, you can use the `highlight=1,3-5` syntax to highlight specific lines in a `pre` code block:

    ```js highlight=1,3-5
    import useSWR from 'swr'
    
    export default function Page() {
      // ...
    }
    ```

**In v2**, you need to wrap it with `""` to make it a [valid JSX prop](https://github.com/remcohaszing/remark-mdx-code-meta):

    ```js highlight="1,3-5"
    import useSWR from 'swr'
    
    export default function Page() {
      // ...
    }
    ```

---

_(The following is generated automatically and includes all PRs since this is the first release created on GitHub)_

## What's Changed
* Convert Windows backslash path to slash path by @sinchang in https://github.com/shuding/nextra/pull/11
* I18n by @shuding in https://github.com/shuding/nextra/pull/19
* dev(nextra): Add watch command by @davesnx in https://github.com/shuding/nextra/pull/38
* fix(loader): JSON parse throwing if isn't valid by @davesnx in https://github.com/shuding/nextra/pull/37
* Edit in Github link support for any random repository directories by @davesnx in https://github.com/shuding/nextra/pull/43
* Add Prettier config by @pacocoursey in https://github.com/shuding/nextra/pull/44
* Fix local example build on core by @pacocoursey in https://github.com/shuding/nextra/pull/45
* Decrease anchor links font size by @pacocoursey in https://github.com/shuding/nextra/pull/46
* Add postcss for Tailwind watch mode by @shuding in https://github.com/shuding/nextra/pull/47
* Ignore files with empty basename, like .DS_Store etc by @fanweixiao in https://github.com/shuding/nextra/pull/62
* Fix/css display property by @gr-qft in https://github.com/shuding/nextra/pull/55
* Add Bleed component by @shuding in https://github.com/shuding/nextra/pull/68
* Accept mdx options by @rot1024 in https://github.com/shuding/nextra/pull/76
* fix: og protocol peropty in meta by @huozhi in https://github.com/shuding/nextra/pull/79
* Change the logo wrapper from inline to block by @thien-do in https://github.com/shuding/nextra/pull/88
* Option for disabling the default font by @shuding in https://github.com/shuding/nextra/pull/89
* Fix error with React 17, i18n routes and SSG/SSR by @shuding in https://github.com/shuding/nextra/pull/96
* Fix search focusing by @madmanwithabike in https://github.com/shuding/nextra/pull/98
* Update Callout component to correctly support emojis by @shuding in https://github.com/shuding/nextra/pull/106
* Make external links safe by default. by @leerob in https://github.com/shuding/nextra/pull/111
* build stork index only when stork:true by @navin-moorthy in https://github.com/shuding/nextra/pull/120
* Allow menus collapsed by default by @benoror in https://github.com/shuding/nextra/pull/122
* minor accessibility issues ul-li by @marcosrjjunior in https://github.com/shuding/nextra/pull/138
* Fix `test.en.mdx` file typo errors, add search result list circular key handle support by @AmagiDDmxh in https://github.com/shuding/nextra/pull/146
* Support Cusdis in nextra-theme-blog by @djyde in https://github.com/shuding/nextra/pull/143
* feat: get current year dynamically by @surjithctly in https://github.com/shuding/nextra/pull/156
* Fixed fresh loading from URLs containing header tags by @nguyenvukhang in https://github.com/shuding/nextra/pull/167
* Support static next/image out of the box by @shuding in https://github.com/shuding/nextra/pull/177
* Fix GFM support and upgrade MDX loader by @shuding in https://github.com/shuding/nextra/pull/184
* Better Callouts by @shuding in https://github.com/shuding/nextra/pull/185
* Prefix stork options with `unstable_` by @shuding in https://github.com/shuding/nextra/pull/186
* Add `unstable_faviconGlyph` option to the docs theme by @shuding in https://github.com/shuding/nextra/pull/187
* Support highlighting a range of code lines by @shuding in https://github.com/shuding/nextra/pull/189
* Normalize option names by @shuding in https://github.com/shuding/nextra/pull/190
* Update loader.js by @olmesm in https://github.com/shuding/nextra/pull/180
* Support page types by @shuding in https://github.com/shuding/nextra/pull/191
* Improvements to search and mobile layout by @shuding in https://github.com/shuding/nextra/pull/192
* Improve navbar and sidebar for the docs theme by @shuding in https://github.com/shuding/nextra/pull/193
* add titleSuffix config to blog theme by @thien-do in https://github.com/shuding/nextra/pull/201
* Fixing routes so they work in Windows by @dzearing in https://github.com/shuding/nextra/pull/215
* Add option to change the Git icon in the top right corner by @maltejur in https://github.com/shuding/nextra/pull/224
* Add `rel="noreferrer"` to all links with `target="_blank"` by @maltejur in https://github.com/shuding/nextra/pull/232
* feat: provide title fallback by @geekplux in https://github.com/shuding/nextra/pull/237
* Stripe-like search input by @jaredpalmer in https://github.com/shuding/nextra/pull/245
* Improve nav bar by @jaredpalmer in https://github.com/shuding/nextra/pull/246
* fix for MDXProvider/TOC, and bump mdx version to 2.0.0-rc2  by @ChinW in https://github.com/shuding/nextra/pull/251
* Fix sidebar for `trailingSlash: true` by @maltejur in https://github.com/shuding/nextra/pull/236
* Feature: dark mode for nextra-theme-blog by @geekplux in https://github.com/shuding/nextra/pull/242
* feat: custom head according to meta or frontMatter of post by @geekplux in https://github.com/shuding/nextra/pull/240

## New Contributors
* @sinchang made their first contribution in https://github.com/shuding/nextra/pull/11
* @davesnx made their first contribution in https://github.com/shuding/nextra/pull/38
* @fanweixiao made their first contribution in https://github.com/shuding/nextra/pull/62
* @gr-qft made their first contribution in https://github.com/shuding/nextra/pull/55
* @rot1024 made their first contribution in https://github.com/shuding/nextra/pull/76
* @huozhi made their first contribution in https://github.com/shuding/nextra/pull/79
* @thien-do made their first contribution in https://github.com/shuding/nextra/pull/88
* @madmanwithabike made their first contribution in https://github.com/shuding/nextra/pull/98
* @leerob made their first contribution in https://github.com/shuding/nextra/pull/111
* @navin-moorthy made their first contribution in https://github.com/shuding/nextra/pull/120
* @benoror made their first contribution in https://github.com/shuding/nextra/pull/122
* @marcosrjjunior made their first contribution in https://github.com/shuding/nextra/pull/138
* @AmagiDDmxh made their first contribution in https://github.com/shuding/nextra/pull/146
* @djyde made their first contribution in https://github.com/shuding/nextra/pull/143
* @nguyenvukhang made their first contribution in https://github.com/shuding/nextra/pull/167
* @olmesm made their first contribution in https://github.com/shuding/nextra/pull/180
* @dzearing made their first contribution in https://github.com/shuding/nextra/pull/215
* @jaredpalmer made their first contribution in https://github.com/shuding/nextra/pull/245
* @ChinW made their first contribution in https://github.com/shuding/nextra/pull/251

**Full Changelog**: https://github.com/shuding/nextra/commits/v2.0.0-beta.2

<hr /><em>This discussion was created from the release <a href='https://github.com/shuding/nextra/releases/tag/v2.0.0-beta.2'>Nextra 2.0.0 (beta.2)</a>.</em>

**Comment by maltejur** [comment]
Does this work with Next.js 12? I am currently trying to set it up and I can't get it to work because of some ES6 stuff.

![image](https://user-images.githubusercontent.com/48161361/143888307-8778e943-81db-4705-82fc-c1e3bbe202fe.png)

**Reply by shuding** [reply]
Yes it should work with Next.js 12. Can you delete `.next` and try again?

**Reply by maltejur** [reply]
I tried to do that and it didn't work. I found a fix, but it is kind of going arround the problem:
```ts
// next.config.js
const withNextra = require("nextra")({
  theme: "nextra-theme-docs",
  themeConfig: "./theme.config.tsx"
});
const withTM = require("next-transpile-modules")([
  "nextra-theme-docs"
]);

module.exports = withTM(
  withNextra({
    reactStrictMode: true,
    trailingSlash: true
  })
);
```
With: \
next: 12.0.4 \
nextra: 2.0.0-beta.2 \
nextra-theme-docs: 2.0.0-beta.2

**Comment by dsousa12** [comment]
The code highlighting not working for me. Any solution?

**Comment by Subham-Maity** [comment]
@dsousa12 

You can do something like this 

https://github.com/shuding/nextra/blob/core/examples/swr-site/pages/docs/getting-started.en-US.mdx?plain=1#L97-L135

`jsx {7-11,17,18,27}`

**Comment by wottpal** [comment]
Is there an official migration guide?
